apps/staking/app/mystakes/modules/ClaimTokensModule.tsx

```diff
@@ -13,110 +15,117 @@ import { externalLink } from '@/lib/locale-defaults';
-                className={cn(
-                  'mb-1 h-7 w-7 transition-all duration-300 motion-reduce:transition-none',
-                  isDisabled
-                    ? 'fill-session-text opacity-50'
-                    : 'fill-session-green group-hover:fill-session-black'
-                )}
-              />
-              {dictionary('title')}
-            </ModuleText>
-          </ModuleContent>
-        </ButtonModule>
-      </AlertDialogTrigger>
-      <AlertDialogContent dialogTitle={dictionary('title')}>
-        <ClaimTokensDialog />
-      </AlertDialogContent>
-    </AlertDialog>
+    <ModuleButtonDialogTrigger
+      dialogContent={address ? <ClaimTokensDialog address={address} /> : null}
+      dialogTitle={dict('title')}
+      label={dict('title')}
+      data-testid={ButtonDataTestId.Claim_Tokens_Open_Dialog}
+      disabled={
+        !(address && canClaim && unclaimedRewards) || isRemoteFlagLoading || isClaimRewardsDisabled
+      }
+      IconComp={iconOverride}
+      textClassName={textClassName}
+      iconStrokeForFill={iconStrokeForFill}
+    />
   );
 }
 
-// biome-ignore lint/complexity/noExcessiveCognitiveComplexity: TODO: simplify this component
-function ClaimTokensDialog() {
+function ClaimTokensDialog({ address }: { address: Address }) {
   const dictionary = useTranslations('modules.claim');
-  const dictionaryFee = useTranslations('fee');
-  const dictionaryDialog = useTranslations('modules.claim.dialog');
-  const dictionaryStage = useTranslations('modules.claim.stage');
+  const { canClaim, refetch } = useUnclaimedTokens({ addressOverride: address });
 
-  const { address, chainId } = useWallet();
-  const { canClaim, refetch, formattedUnclaimedRewardsAmount } = useUnclaimedTokens();
-
-  const {
-    data: rewardsClaimData,
-    isError,
-    isSuccess,
-  } = useStakingBackendQueryWithParams(
+  const { data: claimData, isError } = useStakingBackendQueryWithParams(
     getRewardsClaimSignature,
-    { address: address! },
+    { address },
     {
       enabled: !!address && canClaim,
       staleTime: QUERY.STALE_TIME_CLAIM_REWARDS,
     }
   );
 
-  const [rewards, blsSignature, excludedSigners] = useMemo(() => {
-    if (!rewardsClaimData || !('rewards' in rewardsClaimData) || !rewardsClaimData.rewards) {
-      return [undefined, undefined, undefined];
-    }
-    const { amount, signature, non_signer_indices } = rewardsClaimData.rewards;
+  return (
+    <>
+      {isError ? (
+        <ErrorMessage
+          refetch={refetch}
+          message={dictionary.rich('error')}
+          buttonText={dictionary('errorButton')}
+          buttonDataTestId={ButtonDataTestId.Claim_Tokens_Error_Retry}
+        />
+      ) : claimData && address ? (
+        <ClaimTokens claimData={claimData} address={address} />
+      ) : (
+        <Loading />
+      )}
+    </>
+  );
+}
 
-    return [BigInt(amount), signature, non_signer_indices.map(BigInt)];
-  }, [rewardsClaimData]);
+function ClaimTokens({
```

Comment:
Maybe put this above. I know hoisting makes it fine but... 😂

Reply:
its in render order for clarity tho? you read the file top to bottom

Reply:
Either way this is going into a new file